TypeScript/src/components/Part-A.ts

```diff
@@ -0,0 +1,121 @@
+import { Order } from './Order';
+import sortBy from 'lodash.sortby';
+import { function1PartA } from '../output/Output';
+import promptSync from 'prompt-sync';
+const prompt = promptSync();
+type MaybeNumber = number | null;
+
+const staticOrders = () : Order[] =>{
+    const array = [52, 208, 31, 66, 110, 5, 88, 300];
+    let arrayOrdersStatic = [];
+    for(var i = 0 ; i<array.length; i++){
+        arrayOrdersStatic[i] = new Order(array[i]);     
+    }  
+    return arrayOrdersStatic;  
+}
+
+interface OrderInterface {
+    bagsNumber: MaybeNumber;
+    totalCoffePrice: MaybeNumber;
+    totalLargeBoxes: MaybeNumber;
+    totalMediumBoxes: MaybeNumber;
+    totalSmallBoxes: MaybeNumber;
+    largeBoxesPrice: MaybeNumber;
+    mediumBoxesPrice: MaybeNumber;
+    smallBoxesPrice: MaybeNumber;
+    totalOrderPrice: MaybeNumber;
+}
+
+const orderObjects = () : OrderInterface[] =>{
+    const array = staticOrders();
+    const arrayObjects = [];
+    for (var i = 0;i < array.length; i++){
+        arrayObjects[i] = orderInformation(array[i]);
+    } 
+    return arrayObjects;
+}
+
+const orderInformation = <T extends Order>(obj : T) : OrderInterface  =>{
+    
```

Comment:
can you explain in english what extends does and order interface?

Reply:
this is a method  who will return a Object with type 'OrderInterface' and take a parameter an Object created by the class;
More explaination: <T extends Order> and (obj : T) we are specifying that the parameter is an Object created from the class bt giving it a generic type who is extended from the class Order.

OrderInterface is specifying our method type and saying that we are returning an object with type OrderInterface.

Why I created an OrderInterface?
because I want to create an object containing only the parameters that will be shown in the application, the Order class is too big and containing multiple parameters not necessary to be seen in the application and manipulating them (for example the price of one Coffe Bag);
I am using this way of coding because later if I have a big class (For example class for a User on a website having all the parameters of this user).  If I am writing a code for one page from this website and using some informations abut this user I prefer to create a small object from this big Object (created from this class)  and manipulate it.